enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,522 @@
+manifests will require changes to make them suitable for MicroShift. These manifests will reside
+in MicroShift repository.
+
+Because one of the required IPAMs is the `dhcp` (dynamic), manifests will also include
+a DHCP server DaemonSet.
+
+Updating necessary image references will be part of existing rebase procedure.
+
+#### RPM package
+
+RPM spec to build `microshift-multus` and `microshift-multus-release-info` RPMs will be part of
+existing `microshift.spec` file.
+The RPM will include:
+- manifests required to deploy Multus on MicroShift
+- CRI-O drop-in config to use Multus instead of ovn-kubernetes (which will require reorganization
+  of currently existing MicroShift's CRI-O configs)
+- greenboot healthcheck script
+- cleanup script plugin
+
+#### Container images: Multus and network plugins
+
+Because the Multus image used by OpenShift has a large size that is not acceptable for edge devices, new image
+will be prepared and will only include relevant artifacts such as entrypoint script/binary and
+Multus CNI binary (which is copied to host's `/opt/cni/bin`).
+
+To supply network plugins (CNIs) such as `bridge`, `ipvlan`, and `macvlan` a new image will be prepared
+so MicroShift uses the same binaries as OpenShift (alternative is using RHEL's networkplugins RPM - see alternatives).
+This image will also contain IPAM binaries such as `static`, `dynamic (DHCP)`, and `host-local`.
+
+Both of these images will be part of the OpenShift payload which MicroShift references during rebase procedure.
+
+#### Hypershift [optional]
+
+No, enhancement is MicroShift specific.
+
+### Risks and Mitigations
+
+There may be a race condition between Multus on MicroShift and other services on the host that also
+configure the host's networking
+Taking `bridge` CNI as an example: when the bridge interface does not exist, it will be created
+when a Pod requiring that interface is created.
+If user expects something else will create the interface, they will need to configure system to start
+MicroShift after other services.
+
+Multus using thin plugin architecture creates a kubeconfig when the DaemonSet starts and copies it
+out of Pod to `/opt/cni/bin` on the host filesystem so the Multus CNI binary can use it to get
+NetworkAttachmentDefinitions and Pod annotations. The file is owned and readable only by root, so
+the risk of someone gaining access to the cluster after logging into the host can be compared to
+getting access to `/var/lib/microshift/resources/kubeadmin/kubeconfig`.
+
+Currently MicroShift runs as a single-node cluster but in future releases there might be an effort
+to allow for multi-node clusters. For this reason, during implementation of this enhancement, no
+assumption that MicroShift will always run as a single-node should be made and potential multi-node
+should be kept in mind. We will not build the solution to support multi-node, but want to avoid
+making decisions that make it harder to do so in the future.
```

Comment:
good to know! so there is a possibility whereabouts is needed in the future when multi-node is introduced.